### PR TITLE
Fix/use go mod for go bom generation

### PR DESCRIPTION
**Pull request by sullivtr**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1004,23 +1004,45 @@ const createPythonBom = async (
  * @param callback Function callback
  */
 const createGoBom = async (includeBomSerialNumber, path, options, callback) => {
+  // Read in go.sum and merge all go.sum files.
+  let goSumData = new String("")
```
Comment (cretz) [inline index.js]: `let goSumDate = ""` is clearer

Comment (cretz) [inline index.js]: Inconsistent semicolon use (here and in many places)

File: utils.js [hunk 2]
```diff
@@ -982,18 +978,80 @@ const parseGosumData = async function (gosumData) {
...
     }
   }
-  return pkgList;
+  return pkg;
+};
+exports.getGoPkgComponent = getGoPkgComponent;
+
+const parseGoModData = async function (goModData, goSumData) {
+  const pkgComponentsList = [];
+  let isModReplacement  = new Boolean(false);
```
Comment (cretz) [inline utils.js]: Just using `false` is fine

File: utils.js [hunk 3]
```diff
@@ -982,18 +978,80 @@ const parseGosumData = async function (gosumData) {
...
+  if (!goModData) {
+    return pkgComponentsList;
+  }
+
+  const pkgs = goModData.split("\n");
+  for (let i in pkgs) {
+    const l = pkgs[i];
+
+    // Skip go.mod file headers, whitespace, and/or comments
+    if (l.includes("module ") || l.includes("go ")  || l.includes(")") || l.trim() === "" || l.trim().startsWith("//")){
```
Comment (cretz) [inline utils.js]: Inconsistent spacing around boolean operators

File: utils.js [hunk 4]
```diff
@@ -982,18 +978,80 @@ const parseGosumData = async function (gosumData) {
...
+      // Add group, name and version component properties for required modules
+      let group = path.dirname(tmpA[0]);
+      const name = path.basename(tmpA[0]);
+      if (group === ".") {
+        group = name;
+      }
+      const version = tmpA[1]
+      await getGoPkgComponent(goSumData, group, name, version)
+        .then((component) => {
+          if(Object.keys(component).length !== 0) {
```
Comment (cretz) [inline utils.js]: Inconsistent spacing after `if`

File: utils.js [hunk 5]
```diff
@@ -982,18 +978,80 @@ const parseGosumData = async function (gosumData) {
...
+
+    if (!isModReplacement) {
+      // Add group, name and version component properties for required modules
+      let group = path.dirname(tmpA[0]);
+      const name = path.basename(tmpA[0]);
+      if (group === ".") {
+        group = name;
+      }
+      const version = tmpA[1]
+      await getGoPkgComponent(goSumData, group, name, version)
```
Comment (cretz) [inline utils.js]: If you want you could `const component = await getGoPkgComponent(goSumData, group, name, version)` then add it to the array

Comment (sullivtr) [reply utils.js]: I went back and forth. IMO this is a more idiomatic promise handling paradigm. I am wondering if the upstream reviewers will approve of this

**Pull request closed without merging**